### PR TITLE
remove location_id from primary sql models, save to elasticsearch

### PR DESCRIPTION
**Pull request by czue**
@snopoke cc @gcapalbo 

wait for tests

### CODE REVIEW

File: corehq/pillows/ledger.py [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
+from corehq.util.quickcache import quickcache
 from pillowtop.checkpoints.manager import PillowCheckpoint, PillowCheckpointEventHandler
 from pillowtop.pillow.interface import ConstructedPillow
 from pillowtop.processors.elastic import ElasticProcessor
 from pillowtop.reindexer.change_providers.django_model import DjangoModelChangeProvider
 from pillowtop.reindexer.reindexer import ElasticPillowReindexer
 
 
+
+@quickcache(['case_id'])
```
Comment (dimagimon) [inline corehq/pillows/ledger.py]: E303 too many blank lines (3)

File: corehq/apps/reports/commtrack/standard.py [hunk 2]
```diff
@@ -131,38 +134,37 @@ def product_data(self):
...
-            stock_states = stock_states.filter(sql_product__program_id=self.program_id)
-
         product_grouping = {}
-        for state in stock_states:
-            status = state.stock_category
-            if state.product_id in product_grouping:
-                product_grouping[state.product_id][status] += 1
-                product_grouping[state.product_id]['facility_count'] += 1
+        for ledger_value in ledger_values:
+            consumption_helper = get_consumption_helper_from_ledger_value(Domain.get_by_name(self.domain), ledger_value)
```
Comment (dimagimon) [inline corehq/apps/reports/commtrack/standard.py]: E501 line too long (120 > 115 characters)

File: corehq/pillows/ledger.py [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
+def _location_id_for_case(case_id):
+    try:
+        return SQLLocation.objects.get(supply_point_id=case_id).location_id
+    except ObjectDoesNotExist:
```
Comment (gcapalbo) [inline corehq/pillows/ledger.py]: not a blocker, but just curious why the preference for `ObjectDoesNotExist` over `SQLLocation.DoesNotExist`?

**Comment by gcapalbo** [comment]
looks good to me, will let @snopoke sign off

**Comment by czue** [comment]
:white_check_mark: @snopoke

File: corehq/pillows/ledger.py (hunk 3 above)
Comment (czue) [inline corehq/pillows/ledger.py]: no good reason. https://github.com/dimagi/commcare-hq/pull/12111/commits/8cc1998b1dffac84c775f7b8f5f69a2c18b75662

**Comment by snopoke** [comment]
Can we also remove it from the `StockTransactionHelper` since it doesn't seem to be used anywhere?

**Comment by czue** [comment]
good call - probably. will look.

**Comment by czue** [comment]
(it does how somehow make its way into couch `StockState`s but I think not through the helper.)

**Comment by snopoke** [comment]
Yea, that happens here:
https://github.com/dimagi/commcare-hq/blob/master/corehq/ex-submodules/casexml/apps/stock/signals.py#L99-L99

**Pull request merged**